v7/navstories/navstories.py

```diff
@@ -26,24 +26,146 @@
+        """
+        Map form list of pages going into menu to tuble of format as NAVIGATION_LINKS and NAVIGATION_LINKS_POST_NAVSTORIES
+        List format:
+        - List of "top level entry"
+          - List
+            - Top level Menu text, or None if auto-mapped, i.e. not in NAVSTORIES_MAPPING
+            - List of pages in the top menu entry
+              - List of:
+                - navpath: List containing navigation hieracy (permalink without langinfo (initial /en/) and without the NAVSTORIES_PATHS (e.g. /pages/))
+                - Permalink
+                - Page title
+        Example:
+        [   [   'Menu text for nav',
+                [   [['nav'], '/pages/nav/', 'nav/'],
+                    [['nav', 'p2', 'a'], '/pages/nav/p2/a/', 'Page 2a'],
+                    [['nav', 'P2', 'b'], '/pages/nav/P2/b/', 'Page 2b'],
+                    [['nav', 'P2', 'a'], '/pages/nav/P2/a/', 'Page 2a'],
+                    [['nav', 'p2'], '/pages/nav/p2/', 'Page 2'],
+                    [['nav', 'p1'], '/pages/nav/p1/', 'Side 1']]],
+            [   'Menu text for A',
+                [[['A', 'last'], '/pages/A/last/', 'Page title for A/last']]],
+            [   None,
+                [[['B', 'cde'], '/pages/B/cde/', 'Page title for B/cde']]]]
+        """
+        ret = []
+        for topent in entries:
+            # Determine toplevel menu name
+            # - If not None, use the name
+            # - If None, use title of page if page with navpath length=1 exist, else navpath[0]
+            title = topent[0]
+            if not title:
+                # Default is 1th level of navpath
+                title = topent[1][0][0][0]
+                # Search for toplevel page (navpath length = 1
+                for i in range(len(topent[1])):
+                    if len(topent[1][i][0]) == 1:
+                        title = topent[1][i][2] # Page Title
+            if len(topent[1]) == 1 and len(topent[1][0][0]) == 1:
+                # Only one menu item and it is not a subpage, let the item go direct to top level menu
+                #ret += tuble(topent[1][0][1], title)
+                pass
+            else:
+                sub = []
+                # Find min/max depth in actual submenu
+                min_depth = 100 # High initial value, set to min(len(navpath)) in this submen
+                max_depth = 0 # Low initial value, set to max(len(navpath)) in this submen
+                for p in topent[1]:
+                    min_depth = min(min_depth, len(p[0]))
+                    max_depth = max(max_depth, len(p[0]))
+                # Map pages to submenu
+                for p in sorted(topent[1], key=lambda x: x[1]): # Sort by second element (permalink) in page list
+                    prefix = self.navstories_submenu_indention * (len(p[0]) - min_depth)
+                    sub.append(tuple([p[1], prefix + p[2]]))
+                ret.append(tuple([tuple(sub), title]))
+        return tuple(ret)
```

Comment:
No, it returns a tuple of the same format as  NAVIGATION_LINKS (appended to NAVIGATION_LINKS as the last thing that navstories do)